### PR TITLE
[495975] prevent npe for non source folder resources

### PR DESCRIPTION
**Pull request by cdietrich**
[495975] prevent npe for non source folder resources

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=495975

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/JDTAwareEclipseResourceFileSystemAccess2.java [hunk 1]
```diff
@@ -63,12 +63,14 @@ protected void createContainer(IContainer container) throws CoreException {
 	protected void addToSourceFolders(IContainer container) throws JavaModelException {
 		IJavaProject jp = JavaCore.create(container.getProject());
 		if (jp.exists() && !jp.isOnClasspath(container)) {
-			IPackageFragmentRoot currentSource = jp.findPackageFragmentRoot(jp.getPath().append(getCurrentSource()));
-			if (currentSource != null) {
-				IClasspathEntry currentClasspathEntry = currentSource.getRawClasspathEntry();
-				if (currentClasspathEntry != null) {
-					insertClasspathEntry(container, currentClasspathEntry, jp);
-					return;
+			if (getCurrentSource() != null) {
```
Comment (szarnekow) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/JDTAwareEclipseResourceFileSystemAccess2.java]: A little bit of nitpicking here, but the local variable currentSource and the method getCurrentSource() are potentially confusing for readers. Maybe use IPackageFragmentRoot pckFragmentRoot instead?

Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/JDTAwareEclipseResourceFileSystemAccess2.java]: hmmm i just kept it consistent with other places. but i can refactor that as well

Comment (cdietrich) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/JDTAwareEclipseResourceFileSystemAccess2.java]: updated

Comment (szarnekow) [inline org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/JDTAwareEclipseResourceFileSystemAccess2.java]: thank you

**Pull request merged**